src/dropzone/DropzoneDirective.ts

```diff
@@ -0,0 +1,45 @@
+import * as angular from 'angular';
+
+import Dropzone = require('dropzone');
+
+/**
+ * Define new interface to allow accessing '$ctrl' from scope
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18196
+ */
+interface IDropzoneDirectiveScope extends ng.IScope {
+  $ctrl: any;
+}
+
+export class DropzoneDirective implements ng.IDirective {
+  /**
+   * Match class name.
+   */
+  public restrict: string = 'C';
+
+  /**
+   * Update the DOM to implement the directive.
+   */
+  public link(
+    scope: IDropzoneDirectiveScope,
+    element: ng.IAugmentedJQuery,
+    attrs: ng.IAttributes
+  ) {
+    let dropzoneConfig = {
+      url: scope.$ctrl.config.url
+    };
+
+    let eventHandlers = {
+      'sending': (file, xhr, formData) => {
+        // append AWS upload key to the form data
```

Comment:
It's not a big deal because the reference is in a comment but keep in mind this code should work with all backends.

Reply:
Yeah, I agree as well. I know currently we only have the S3 backend defined and the 'key' form data is required to make POSTs on S3. I'll update the comment here to make note this is specific for AWS S3 (and if we have additional backends, we may need to pass that info to the dropzone controller as well).